### ISSUE TITLE
Meson crashes when using find_library in declare_dependency

### ISSUE THREAD

**Issue by rnburn** [mesonbuild/meson#2727]
Using version meson-0.43.0 with

meson_options.txt:
```
option('with_tbb', type: 'string', value: '')
```
meson.build:
```
project('tutorial', 'cpp')
with_tbb = get_option('with_tbb')
cxx = meson.get_compiler('cpp')

libtbb = cxx.find_library('tbb', dirs:join_paths(with_tbb, 'lib'))
tbb = declare_dependency(
  include_directories: include_directories(with_tbb + '/include'),
  link_with: [libtbb]
)

executable('demo', 'main.cpp', dependencies : [tbb])
```
If I run `meson -D with_tbb=/usr/local builddir`, I get
```
The Meson build system
Version: 0.43.0
Source dir: /Users/ryanburn/test/meson
Build dir: /Users/ryanburn/test/meson/builddir
Build type: native build
Project name: tutorial
Native C++ compiler: c++ (clang 8.1.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Library tbb found: YES
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 352, in run
    app.generate()
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 147, in generate
    self._generate(env)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 195, in _generate
    intr.run()
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreter.py", line 2785, in run
    super().run()
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 148, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 170, in evaluate_codeblock
    raise e
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 164, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 175, in evaluate_statement
    return self.function_call(cur)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 387, in function_call
    return self.funcs[func_name](node, self.flatten(posargs), kwargs)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreterbase.py", line 77, in wrapped
    return f(s, node_or_state, args, kwargs)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreter.py", line 2196, in func_executable
    return self.build_target(node, args, kwargs, ExecutableHolder)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/interpreter.py", line 2889, in build_target
    target = targetclass(name, self.subdir, self.subproject, is_cross, sources, objs, self.environment, kwargs)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/build.py", line 1137, in __init__
    super().__init__(name, subdir, subproject, is_cross, sources, objects, environment, kwargs)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/build.py", line 358, in __init__
    self.process_kwargs(kwargs, environment)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/build.py", line 706, in process_kwargs
    self.add_deps(deplist)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/build.py", line 832, in add_deps
    self.link(l)
  File "/usr/local/lib/python3.6/site-packages/mesonbuild/build.py", line 866, in link
    if not t.is_linkable_target():
AttributeError: 'ExternalLibrary' object has no attribute 'is_linkable_target'
```
Not sure if this way of declaring a dependency is supported, but would expect to at least get an error message instead of crashing in python code.

**Comment by nirbheek** [comment]
We should probably make `link_with:` and `dependencies:` synonyms on the user-facing end.

**Comment by eli-schwartz** [comment]
This was actually fixed shortly before it was reported in #2473

However, it was only fixed in git, and later in meson-0.44.0, so the release version that the original report used was too old to have that safeguard...
